db/migrate/20170602111120_add_timestamps_to_documents.rb

```diff
@@ -0,0 +1,110 @@
+class AddTimestampsToDocuments < ActiveRecord::Migration[5.1]
+  disable_ddl_transaction!
+
+  def change
+    add_timestamps(:documents, null: true) unless column_exists?(:documents, :created_at)
+    puts "> Timestamp fields added."
+    puts "> Updating multiple-locale documents without editions..."
+
+    update_multiple_locale_documents_without_editions
+    puts "> Multiple-locale documents without editions updated."
+    puts "> Updating remaining documents..."
+
+    Document.where(created_at: nil).find_each do |document|
+      editions = Edition.where(document_id: document.id).order(:user_facing_version)
+      if editions.exists?
+        update_timestamps_from_editions(document, editions)
+      else
+        update_timestamps_from_events(document)
+      end
```

Comment:
I originally used `puts '.'` but canned it as it the output was so long! But I agree that feedback is useful, and a count seems more useful than a full stop :)